libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationTelemetry.java

```diff
@@ -560,8 +562,13 @@ private BatteryEvent buildBatteryEvent() {
     BatteryMonitor batteryMonitor = new BatteryMonitor(currentSdkVersionChecker);
     float batteryPercentage = batteryMonitor.obtainPercentage(context);
     boolean isPluggedIn = batteryMonitor.isPluggedIn(context);
-    return new BatteryEvent(navigationSessionState.sessionIdentifier(), batteryPercentage,
-      isPluggedIn, performanceMetadata);
+    String sessionId = navigationSessionState.sessionIdentifier();
+    if (sessionId != null) {
```

Comment:
Is this `null` check necessary? `navigationSessionState.sessionIdentifier()` is guaranteed to be non-null 👀 https://github.com/mapbox/mapbox-navigation-android/blob/6f91e57fae21ceb8a895b4a8f95fac400c6ce6e9/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationTelemetry.java#L161

Although it's not marked as so in https://github.com/mapbox/mapbox-navigation-android/blob/6f91e57fae21ceb8a895b4a8f95fac400c6ce6e9/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/metrics/SessionState.java#L107 (we should add `@NotNull` annotation)

Reply:
In any case I think it's wrong, if it's equals to `null` we should return `null`, otherwise create the event 😬

Reply:
Yeah, you are right, I missed it. And weirdly failed with true\false blocks 🤕